### PR TITLE
Refactor `DashboardViewController` to have one top-level scroll view instead of multiple scroll views for stats

### PR DESCRIPTION
**Pull request by jaclync**
Main part of #8911 

This is a big change on the dashboard, I'm targeting release 12.7 to make time for review and testing.

## Why

With the addition of the onboarding card UI, the header view (announcement cards, onboarding cards, onboarding card, etc.) is taking up more vertical space. With the current dashboard implementation, the header view is not scrollable, only the tabbed stats view is scrollable in 4 separate tabs. When the header view is taking significant vertical space like in larger font sizes, it can be hard or even impossible to scroll up to see the stats. In this PR, the dashboard UI hierarchy was refactored so that it contains:

- Stack view (`DashboardViewController.stackView`)
  - Scroll view (`DashboardViewController.containerView`) - this should be the only scroll view in the UI hierarchy
    - Stack view (`DashboardViewController.containerStackView`)
      - Header view (store name label, announcement card, add product card, onboarding card, etc.) (`DashboardViewController.headerStackView`)
      - Stats view (`DashboardViewController.dashboardUI` - `StoreStatsAndTopPerformersViewController` or `DeprecatedDashboardStatsViewController`)
  - Jetpack benefits banner (optionally shown)

## UI changes

The main UI change in the dashboard is the refresh control. The spinner used to be under each stats time range tab, and now it's attached to the one-and-only scroll view above the store name label.

## Challenges

There were several challenges I faced in the UI hierarchy refactoring in order to keep the existing functionalities:
- Large title: the refresh control is shown in the navigation bar by default since we're using large title
- Show/hide the header view when the large title is shown/collapsed: currently, we [observe the navigation bar height to hide the header view when the large navigation bar title is collapsed](https://github.com/woocommerce/woocommerce-ios/blob/27f19c59c07a8001f8b57d5093c27e34af122a36/WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift#L744-L756). Now that both the header and stats views are inside the same scroll view / stack view, hiding the header view is tricky because it affects the scroll view content size and this also affects the internal large title state (it reacts to the scroll view state). If we hide the header view and the dashboard content is shorter than the scroll view, the scroll view also becomes not scrollable and the merchant cannot scroll to see the header view anymore. Unfortunately there's no delegate/API to know when large title is shown or collapsed, so this navigation bar frame observation results in buggy scrolling behavior when the deprecated stats view is shown since it's shorter than the scroll view height
  - Workaround: in `DashboardViewController.updateHeaderVisibility`, it doesn't hide/show the header view if the dashboard content height is smaller than the scroll view height. I've tried a few other things like adding a bottom spacer view with a dynamic height to fill the scroll view height but they were very hacky and didn't work well. Please share any ideas! 

## Description

Main changes:
- Removed `DashboardUI.DashboardUIScrollDelegate` since we don't need the large title workaround anymore now that there's only one scroll view
- `usageTracksEventEmitter.interacted` in scroll view scrolling interaction and top performer tap interaction was moved to the new places
- In `DashboardViewController`, the UI hierarchy was refactored based on the description above. Instead of manual Auto Layout constraints that are hard to follow, stack views were used
- Refactored `TopPerformerDataViewController` from a `UITableView` (a `UIScrollView subclass) to use the `TopPerformersView` SwiftUI view (the xib will be removed separately)
- In `DeprecatedDashboardStatsViewController`, the scroll view and its refresh control were removed so that we don't have nested scroll views
- In `StoreStatsV4PeriodViewController.xib`, some vertical constraints were updated so that the height isn't ambiguous

## Testing instructions

Since this affects the dashboard UI and it's hard to feature flag these changes, there are a few scenarios to test:

### Site with dashboard stats

Prerequisite: a site with analytics enabled

- Log in to the JCP site if needed --> the My store tab should look good as before, with the change on the refresh control
- Pull down to refresh --> the data for the visible tab should be refreshed
- Scroll up --> the large title should be shrunken, and the header view should be hidden
- Tap on a different time range tab --> the data should be loaded after a bit
- Tap on a top performer row --> the product form should be shown

### Site with stats disabled

Prerequisite: a site with analytics disabled - you can enable/disable analytics at the site's `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` - it might be easier with self-hosted sites

- Log in to the site in the prerequisite if needed --> the My store tab should look good as before, the deprecated stats view should be shown below the header view
- In the site's `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`, re-enable analytics
- Go back to the app
- Pull down to refresh --> the stats should be shown again

### Jetpack benefits banner

Prerequisite: a JCP site and the Jetpack benefits banner hasn't been dismissed

- Log in to the JCP site if needed --> the Jetpack benefits banner should be shown, and the rest of the dashboard UI should look good
- Tap on the Jetpack benefits banner --> the modal should be shown like before
- Dismiss the modal
- Rotate the device --> the banner should still look good

## Screenshots

light - pull-to-refresh | dark - pull-to-refresh | dark | deprecated stats
-- | -- | -- | --
Simulator Screen Shot - iPhone 14 - 2023-03-03 at 15 53 35 | Simulator Screen Shot - iPhone 14 - 2023-03-03 at 15 53 53 | Simulator Screen Shot - iPhone 14 - 2023-03-03 at 15 53 59 | Simulator Screen Shot - iPhone 14 - 2023-03-03 at 15 54 10

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by pmusolino — APPROVED** [review]
In general, it looks good. 👍 

I just noticed a few issues/improvements.
When the stats are enabled and the device is in landscape mode, I'm not able to see the onboarding list, instead if the stats are disabled, everything works fine. 




Another thing, that could be a UX improvement, is that there is a sort of swipe and not scroll between the onboarding list and stats. I recorded a video that shows it:

https://user-images.githubusercontent.com/495617/224054921-1f9459a9-6c98-47f2-8c4a-29f2d036e2bb.mp4

**Comment by jaclync** [comment]
Thanks a lot for taking the time to review & test this PR @pmusolino!

> I just noticed a few issues/improvements. When the stats are enabled and the device is in landscape mode, I'm not able to see the onboarding list, instead if the stats are disabled, everything works fine.
> 
> 
> Another thing, that could be a UX improvement, is that there is a sort of swipe and not scroll between the onboarding list and stats. I recorded a video that shows it:

These are actually from the pre-existing UX in production, where we hide the header (like the announcement card in the screencast below) when the navigation bar is collapsed - in landscape mode and when the large title is shrunk:


https://user-images.githubusercontent.com/1945542/224227780-7c0538e0-8bce-445b-81e1-9250582c1f5c.MP4

I assumed the onboarding list should behave the same as other elements in the header like the announcement card. Do you think we always want to show the onboarding list regardless of the orientation and large title state? I can make this update in a separate PR.

**Comment by pmusolino** [comment]
Interesting, thank you Jaclyn for sharing it!

I think a better UX could be to scroll the view without that gesture, because it seems we are going to collapse it, instead we are not. It makes sense for a banner, not sure if it's a good user experience for people looking at the onboarding list.

About the onboarding list not being visible in the landscape, I think it should be always visible. It is not a problem for those who use an iPhone, but mainly for those who use an iPad, which I think is used more in landscape mode.

Anyway, feel free to add these improvements in future PR :)

**Comment by jaclync** [comment]
> I think a better UX could be to scroll the view without that gesture, because it seems we are going to collapse it, instead we are not. It makes sense for a banner, not sure if it's a good user experience for people looking at the onboarding list.
> 
> About the onboarding list not being visible in the landscape, I think it should be always visible. It is not a problem for those who use an iPhone, but mainly for those who use an iPad, which I think is used more in landscape mode.

Yea this UX makes sense to be, let's go with this 👍 I will work on this as a separate subtask for https://github.com/woocommerce/woocommerce-ios/issues/8911. I intentionally saved the merge for early in the new release cycle to allow additional fixes/changes.

**Comment by jaclync** [comment]
Merging this now after updating the release notes in https://github.com/woocommerce/woocommerce-ios/pull/9031/commits/9c8ad508bc26aea8ad00ed40c772ed58ef7a746c and pulling the latest `trunk`. cc @selanthiraiyan about this change, hopefully the onboarding UI is not truncated in shorter devices without breaking changes

**Comment by selanthiraiyan** [comment]
> about this change, hopefully the onboarding UI is not truncated in shorter devices without breaking changes

@jaclync Just wanted to report that the store name is truncated in `trunk`. Please ignore if this is WIP.

**Comment by jaclync** [comment]
> Just wanted to report that the store name is truncated in `trunk`. Please ignore if this is WIP.

Thanks for reporting this, I added a subtask to https://github.com/woocommerce/woocommerce-ios/issues/8911. I've also seen store name truncated before the scroll view change, but might be a different issue now https://github.com/woocommerce/woocommerce-ios/issues/8499
